### PR TITLE
[#167727319] Admin assign user role feature

### PR DESCRIPTION
**Pull request by shonubijerry**
### What does this PR do?

* Admin can assign role to a user and also set permissions to roles

### Description of Task to be completed?

* Setup role and permission models
* Add seed database with predetermined roles
* Add seed database with predetermined permissions
* Create checkPermissions middleware to validate user access to a particular route
* Create assign roles route
* Add editUserRole method to user controller and validate data supplied
* Create permissions module
* Write integration test
* Write swagger documentation

### How should this be manually tested?

* Clone the repository, cd into the repository directory, and run `npm install` finally run `npm run start:dev`
  **To view all roles**
  `GET http://localhost:3000/api/v1/roles`
  **To view all permissions**
	`GET http://localhost:3000/api/v1/permissions`
  **To view all role permissions**
	`GET http://localhost:3000/api/v1/roles/:roleId/permissions `
   **To assign role to a user**
	`PATCH http://localhost:3000/api/v1/users/roles/:roleId`
	**To chage role permissions**
  `PATCH http://localhost:3000/api/v1/roles/:roleId/permissions`

### Any background context you want to provide?

* Database structure of the project was significaltly altered for efficiency of the application as regards this feature and subsequent features.

**Relevant pivotal tracker stories?**
#167727319

### CODE REVIEW

File: src/helpers/JWT.js [hunk 1]
```diff
@@ -3,15 +3,21 @@ import { SECRET_KEY, EXPIRATION_DURATION } from '../config/constants';
...
    */
-  static async generateToken(data) {
-    const token = await jwt.sign({ key: data }, SECRET_KEY, { expiresIn: EXPIRATION_DURATION });
+  generateToken: (data) => {
+    const token = jwt.sign({ key: data }, SECRET_KEY, { expiresIn: EXPIRATION_DURATION });
     return token;
-  }
-}
-export default JWT;
+  },
```
Comment (orley12) [inline src/helpers/JWT.js]: Just a suggestion, since we agreed to use functional style. Exporting object literals, is that not just another syntax for the object class style.

**Review by orley12 — COMMENTED** [review]
LGTM

File: src/helpers/JWT.js (hunk 1 above)
Comment (shonubijerry) [reply src/helpers/JWT.js]: I will look into this

File: .babelrc [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "@babel/preset-env"
   ],
   "plugins": [
-    "@babel/plugin-proposal-object-rest-spread",
-    "@babel/transform-runtime"
+    "@babel/transform-runtime",
+    "@babel/plugin-proposal-object-rest-spread"
```
Comment (Hector101) [inline .babelrc]: This change isn't necessary, revert this change.

**Review by Hector101 — CHANGES_REQUESTED** [review]

File: .babelrc (hunk 2 above)
Comment (shonubijerry) [reply .babelrc]: Yes

**Review by steelze — COMMENTED** [review]
LGTM

**Review by steelze — DISMISSED** [review]

**Review by steelze — DISMISSED** [review]
LGTM

**Review by Hector101 — APPROVED** [review]
